pkg/mimirtool/commands/analyse_grafana.go

```diff
@@ -49,26 +49,40 @@ func (f folderTitles) IsCumulative() bool {
 }
 
 func (cmd *GrafanaAnalyzeCommand) run(k *kingpin.ParseContext) error {
+	output, err := AnalyzeGrafana(cmd.address, cmd.apiKey, cmd.folders, cmd.readTimeout)
+	if err != nil {
+		return err
+	}
+	err = writeOut(output, cmd.outputFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AnalyzeGrafana analyze grafana's dashboards and return the list metrics used in them.
+func AnalyzeGrafana(address, apiKey string, folders []string, readTimeout time.Duration) (*analyze.MetricsInGrafana, error) {
```

Comment:
These args seem very unrelated to each other; wdyt about receiving a `(ctx context.Context, c *sdk.Client, folders []string)`

This might be on the bikeshedding side, but let me know what you think.